Scripts/Runtime/Entities/Lifecycle/PrototypeVariant.cs

```diff
@@ -0,0 +1,75 @@
+using Anvil.Unity.DOTS.Core;
+using Anvil.Unity.DOTS.Util;
+using System;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    public readonly struct PrototypeVariant : IEquatable<PrototypeVariant>,
+                                              IComponentData,
+                                              IToFixedString<FixedString32Bytes>
+    {
+        public static readonly int UNSET_VALUE = default;
+
+        public static PrototypeVariant FromEnum<T>(T enumValue)
```

Comment:
🚲 Based on our discussions and leaving room for potential mod support in the future we may just want to pass in raw ints that we grab from the statics on the definitions and use an ID provider.

Reply:
I've added a TODO for it and linked back to this comment. 

I think enum's are nicer to work with right now and we still allow for mod support with the int constructor.  Or alternative static API proposed here: https://github.com/decline-cookies/anvil-unity-dots/pull/271/files#r1265728407